interfaces/cython/cantera/func1.pyx

```diff
@@ -87,3 +100,78 @@ cdef class Func1:
+    or a single array that concatenates those inputs in two columns. Between
+    sample points, values are evaluated based on the keyword argument
+    'interpolation'; options are 'linear' (linear interpolation, default) or
+    'previous' (nearest previous value). Outside the sample interval, the value
+    at the closest end point is returned.
+
+    Examples for `Tabulated1` objects are::
+
+        >>> t1 = Tabulated1([[0, 2], [1, 1], [2, 0]])
+        >>> [t1(v) for v in [-0.5, 0, 0.5, 1.5, 2, 2.5]]
+        [2.0, 2.0, 1.5, 0.5, 0.0, 0.0]
+
+        >>> t2 = Tabulated1([[0, 2], [1, 1], [2, 0]], interpolation='previous')
+        >>> [t2(v) for v in [-0.5, 0, 0.5, 1.5, 2, 2.5]]
+        [2.0, 2.0, 2.0, 1.0, 0.0, 0.0]
+
+        >>> t3 = Tabulated1([0, 1, 2], [2, 1, 0])
+        >>> [t3(v) for v in [-0.5, 0, 0.5, 1.5, 2, 2.5]]
+        [2.0, 2.0, 1.5, 0.5, 0.0, 0.0]
+
+        >>> t4 = Tabulated1(np.array([0, 1, 2]), (2, 1, 0))
+        >>> [t4(v) for v in [-0.5, 0, 0.5, 1.5, 2, 2.5]]
+        [2.0, 2.0, 1.5, 0.5, 0.0, 0.0]
+    """
+
+    def __init__(self, *args, **kwargs):
+        if len(args) == 1:
+            # tabulated function (single argument)
+            arr = np.array(args[0])
+            if arr.ndim == 2:
+                if arr.shape[1] == 2:
+                    time = arr[:, 0]
+                    fval = arr[:, 1]
+                    method = kwargs.get('interpolation', 'linear')
+                    self._set_tables(time, fval, stringify(method))
+                else:
+                    raise ValueError(
+                        "Invalid dimensions: specification of "
+                        "tabulated function with a single array "
+                        "requires two columns")
+            else:
+                raise TypeError(
+                    "Tabulated1 must be constructed from "
+                    "a numeric array with two columns")
+
+        elif len(args) == 2:
+            # tabulated function (two arrays mimic C++ interface)
+            time, fval = args
+            method = kwargs.get('interpolation', 'linear')
+            self._set_tables(time, fval, stringify(method))
+
+        else:
+            raise ValueError("Invalid number of arguments")
+
+    cpdef void _set_tables(self, time, fval, string method) except *:
+        tsiz = np.asarray(time).size
+        fsiz = np.asarray(fval).size
+        if tsiz != fsiz:
+            raise ValueError("Sizes of arrays do not match "
+                             "({} vs {}).".format(tsiz, fsiz))
+        elif tsiz == 0:
+            raise ValueError("Arrays must not be empty.")
+        cdef np.ndarray[np.double_t, ndim=1] tvec = np.asarray(time,
+                                                               dtype=np.double)
+        cdef np.ndarray[np.double_t, ndim=1] fvec = np.asarray(fval,
+                                                               dtype=np.double)
```

Comment:
Since you turn these into arrays on the first two lines of this function, why not just do that once?

Reply:
The `tsiz = 0` exception was necessary to handle empty arrays. I can simplify though ...